lazy-pages/src/sys.rs

```diff
@@ -67,21 +69,19 @@ pub unsafe fn user_signal_handler(info: ExceptionInfo) -> Result<(), Error> {
 
     let mem = info.fault_addr;
     let native_page = region::page::floor(mem) as usize;
-    let wasm_mem_begin = WASM_MEM_BEGIN.with(|x| *x.borrow()) as usize;
-
-    if wasm_mem_begin == 0 {
-        return Err(Error::WasmBeginIsNotSet);
-    }
+    let wasm_mem_begin = LAZY_PAGES_CONTEXT
+        .with(|ctx| ctx.borrow().wasm_mem_addr)
+        .ok_or(Error::WasmMemAddrIsNotSet)? as usize;
 
     if wasm_mem_begin > native_page {
-        return Err(Error::UnknownMemory {
+        return Err(Error::SignalFromUnknownMemory {
             wasm_mem_begin,
             native_page,
         });
     }
 
     // First gear page which must be unprotected
-    let gear_page = PageNumber::from(((native_page - wasm_mem_begin) / gear_ps) as u32);
+    let gear_page = PageNumber(((native_page - wasm_mem_begin) / gear_ps) as u32);
```

Comment:
Use saturating operations here

Reply:
Just 5 strings above we checks that native page >= wasm mem begin